lib/endpoint.js

```diff
@@ -25,9 +25,16 @@ EndpointState.prototype.init = function () {
     this.open_requests = 0;
     this.close_requests = 0;
     this.initialised = false;
+    this.marker = undefined;
+};
+
+EndpointState.prototype.mark = function (o) {
```

Comment:
For consistency can you update the type definition for endpoint in endpoint.d.ts to have marker property and signature of the `mark()` method. A brief description of the purpose of mark() method would also be helpful.

Reply:
done